### DISCUSSION TITLE
Support different relationship interfaces

### DISCUSSION THREAD

**Discussion by flipsasser** [directus/directus#7164] [Feature Requests]
Hello! Okay, a few things to get out of the way:

1. Oh my god, I love Directus
2. Oh my god, thank you for Directus
3. Oh my god... I think I love... you?

Okay. Now that I'm done gushing. Here's a proposal I wanted to make: it would be great to hide some of the implementation details of a join model (i.e. the join table between a many-to-many relationship that may or may not have extra metadata; I don't know what it's called w/y'all but we call it a "join model" which I think is a Rails convention so ... YMMV with the garbage words I'm using) behind a few different interfaces.

USE CASE TIME!

We love Directus because it makes it easy to do stuff like segregate content between production and staging. It lets us use a single interface for managing content while ensuring draft stuff doesn't leak out but we can still preview it with our UI.

We do that with a "hidden" record type called "Environment". "Environment" has a unique `name` column, which is a string, as its primary key. Like, no IDs, no nothin - just the string column.

We then allow various _other_ data models to have a many-to-many relationship with environments. This allows us to let the marketing folks check off a "staging" and "production" environment when they're building out content, and also allows us to add an environment once and have it available for everyone else.

Here's the rub: we can't just give them a checkbox list, like you can with a JSON array column. It would be super duper helpful if we could give them a checkbox list, because the extra "flyaway modal loading up all the environments" is, unfortunately for this use case, overkill.

**TL;DR** different interfaces for relationships would be welcomed but hey, I know you're busy and I'm grateful for Directus regardless. Have a wonderful day.

**Comment by rijkvanzanten** [comment]
Duplicating the existing checkboxes interface, and having it pull it's options from a given collection instead of a hardcoded set in the options sounds doable! Want to give it a shot in a PR?

**Reply by flipsasser** [reply]
I suppose I do.

**Comment by flipsasser** [comment]
Just an update from me: this hasn't fallen off my radar, but is slightly lower priority while we continue to migrate off our old CMS.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
